source/fundamentals/builders/sort.txt

```diff
@@ -11,3 +11,326 @@ Sorts Builders
+
+Text Search
+-----------
+
+If you perform a text search on your collection with the
+:manual:`$text </reference/operator/query/text/>` query operator, you 
+can sort your results by how well they match your ``$search`` String using
+:manual:`text score </reference/operator/aggregation/meta/#exp._S_meta>`. 
+To sort your search results by the text scores associated with your text
+search, use the ``Sorts.metaTextScore()`` static factory method. 
+
+.. warning:: Make Sure to Create a Text Index
+
+   The ``$text`` query operator performs a text search on the content indexed 
+   by the :manual:`text index </core/index-text/>` of your collection.
+   Therefore, you must create a text index on your collection before you
+   can run a text search and use ``Sorts.metaTextScore()``.
+
+.. example::
+
+   In the following code example, we show how you can use the
+   ``Sorts.metaTextScore()`` method to sort the results of a text
+   search on the :ref:`sort_example collection <sorts-builders-sort-example>`.
+   The code example uses the :doc:`Filters </fundamentals/builders/filters>`,
+   :doc:`Indexes </fundamentals/builders/indexes>`, and
+   :doc:`Projections </fundamentals/builders/projections>` builders.
+   The code example performs the following actions:
+
+   #. Creates a text index for your
+      :ref:`sort_example collection <sorts-builders-sort-example>`
+      on the ``food`` field.
+   #. Runs your text search for the phrase "maple donut".
+   #. Projects ``Projections.metaTextScore()`` into your query results as the
+      ``score`` field. This projection is optional if your MongoDB instance is
+      running MongoDB 4.4 or later. 
+   #. Sorts using ``Sorts.metaTextScore()`` as your sort criteria.
+
+   .. code-block:: java
+   
+      import com.mongodb.client.model.Sorts;
+      import com.mongodb.client.model.Projections;
+      import com.mongodb.client.model.Filters;
+      import com.mongodb.client.model.Indexes;
+
+      // <MongoCollection setup code here>
+      
+      collection.createIndex(Indexes.text("food"));
+      Bson metaTextScoreSort = Sorts.metaTextScore("score");
+      Bson metaTextScoreProj = Projections.metaTextScore("score");
+      String searchTerm = "maple donut";
+      Bson searchQuery = Filters.text(searchTerm);
+      collection.find(searchQuery)
+              .projection(metaTextScoreProj)
+              .sort(metaTextScoreSort)
+              .into(results);
+      for (Document result : results) {
+          System.out.println(result.toJson());
+      }
+   
+   The output of the code snippet above should look something like this: 
+   
+   .. code-block:: json
+
+      {"_id": 6, "letter": "c", "food": "maple donut", "score": 1.5}
+      {"_id": 2, "letter": "a", "food": "donuts and coffee", "score": 0.75}
+      {"_id": 3, "letter": "a", "food": "maple syrup", "score": 0.75}
+
+.. note:: MongoDB 4.4 ``$meta`` Behavior
+
+  As of MongoDB 4.4, projecting ``Projections.metaTextScore()`` into
```

Comment:
Thank you for the note. On second reading, I agree that version numbers should not be discussed in the context of time. Fix: https://github.com/mongodb/docs-java/pull/59/commits/7dbe6235e256d8a03953b4cd164e2d435ca104e8

Going forward, I will try to discuss versions in the language of use rather than time ("what are you using" rather than "when are you using something".